### DISCUSSION TITLE
MuPIF installation issue on Windows OS

### DISCUSSION THREAD

**Discussion by solidkon** [mupif/mupif#47] [Q&A] [answered]
I'm trying to install MuPIF distribution on a Windows 10 pc. I have followed the instructions provided in https://mupif.readthedocs.io/en/latest/userman.html#platform-installation but I was not able to download it successfully. Can you please help me solve this? 

I get these errors by trying options 1 and 2 explained in users manual:

1) I would like to let you know that the first option, i.e. 'pip install mupif' does not work at all (this occurred to me as well). It raises this error:
   ```
   ERROR: Could not find a version that satisfies the requirement mupif (from versions: none)
   ERROR: No matching distribution found for mupif
   ```

2) The second option 'pip3 install --upgrade git+https://github.com/mupif/mupif.git' results in the error:
  ```
  Running command git clone --filter=blob:none --quiet https://github.com/mupif/mupif.git 'C:\Users\kkord\AppData\Local\Temp\pip-req-build-5pmcpcyi'
  Resolved https://github.com/mupif/mupif.git to commit c40c07253e06f5219f97255dfc1d4126f33f5303
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [8 lines of output]
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "C:\Users\kkord\AppData\Local\Temp\pip-req-build-5pmcpcyi\setup.py", line 64, in <module>
          long_description=open('README.txt', 'r').read(),
        File "C:\Users\kkord\anaconda3\lib\encodings\cp1253.py", line 23, in decode
          return codecs.charmap_decode(input,self.errors,decoding_table)[0]
      UnicodeDecodeError: 'charmap' codec can't decode byte 0x9c in position 1065: character maps to <undefined>
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
```

**Comment by eudoxos** [comment, marked as answer]
Thank you for contacting us with this. We are in the process of updating the installation instructions in a few different places. What you post is a result of an old installation method (`pip3 install mupif`) which is no longer supported; we disabled it to prevent people from installing ancient version, and to be more flexible supporting several projects using MuPIF. I just fixed the docs you link, could you re-try [Platform Installation](https://mupif.readthedocs.io/en/latest/userman.html#platform-installation)? You want to install the `musicode` branch. Let me know if the installation instructions there are unclear (or broken) so that I can fix them quickly. Sorry for inconveniences.

**Reply by kkordos** [reply]
Hi , i have the same issue. Although, i typed the command 
`git clone --branch musicode https://github.com/mupif/mupif.git` 
on conda command prompt , and i get this error as response:

> Cloning into 'mupif'...
> fatal: Remote branch musicode not found in upstream origin

Can you help me at this one ? Thank you

**Reply by eudoxos** [reply]
Thanks for reporting; I double-checked and see that [the branches](https://github.com/mupif/mupif/branches) and they are capitalized; `--branch Musicode` should do the job.

**Reply by kkordos** [reply]
It works! Thank you

**Reply by solidkon** [reply]
Dear eudoxos thank you for your prompt support. I confirm that the installation proceeds normally now!

**Reply by eudoxos** [reply]
Good, I will close this topic. Feel free to re-open, or open a new one.

**Comment by kkordos** [comment]
Hello guys, i just performed a fresh installation of mupif because the python editor (spyder) could not find mupif. This includes these steps on anaconda prompt with administration privileges:

```
git clone --branch Musicode https://github.com/mupif/mupif.git
cd mupif
pip install -e .
```

 and i get this error again:
```
Obtaining file:///C:/Windows/System32/mupif
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [8 lines of output]
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "C:\Windows\System32\mupif\setup.py", line 64, in <module>
          long_description=open('README.txt', 'r').read(),
        File "C:\Users\kkord\anaconda3\lib\encodings\cp1253.py", line 23, in decode
          return codecs.charmap_decode(input,self.errors,decoding_table)[0]
      UnicodeDecodeError: 'charmap' codec can't decode byte 0x9c in position 1065: character maps to <undefined>
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
```

**Comment by eudoxos** [comment]
Hi, please try again. This was a Windows-specific issue with non-ASCII (?) chars in README.txt. I removed reading that file in #51  from setup.py and it should work now. Let me know whether it is fixed. Thanks.

**Reply by kkordos** [reply]
Yes now it works. Thanks.
